Assets/Project/Modules/Enemies/GeneralEnemyScripts/DealDamageScript.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DealDamageScript : MonoBehaviour
+{
+    private DamageHit _contactDamageHit;
+    [SerializeField] private Transform _transform;
+    [SerializeField] private float _contactHitDamageAmount;
+    [SerializeField] private float _contactHitStunDuration;
+    [SerializeField] private float _contactHitKnockbackForce;
+
+    private void Awake()
+    {
+        _contactDamageHit = new DamageHit(CombatManager.Instance.DamageOnlyPlayerPreset,
+            _contactHitDamageAmount, _contactHitKnockbackForce, _contactHitStunDuration);
+    }
+
+    private void OnTriggerEnter(Collider other)
+    {
+        _contactDamageHit.Position = _transform.position;
+        _contactDamageHit.KnockbackDirection = PositioningHelper.Instance.GetDirectionAlignedWithFloor(_transform.position, other.transform.position);
+        CombatManager.Instance.TryDealDamage(other.gameObject, _contactDamageHit, out DamageHitResult damageHitResult);
+    }
+}
```

Comment:
I also made this script in my current branch XD.

Reply:
Should I delete it, or should I leave it and address the issue after the merge?

Reply:
Don't worry, leave it as it is and I will adress it when merging